test/quantization/fx/test_equalize_fx.py

```diff
@@ -135,220 +210,296 @@ def test_input_weight_eq_observer(self, input_qdtype, input_qscheme, weight_qdty
+            inp_counter = 0
+            weight_counter = 0
+            for node in convert_ref.graph.nodes:
+                if "weight" not in node.name and node.op == 'call_module' and \
+                   isinstance(modules[str(node.target)], MinMaxObserver):
+                    # Check min/max values of input activation layers
+                    exp_min_val, exp_max_val = exp_inp_act_vals[inp_counter]
+                    self.assertEqual(modules[str(node.target)].min_val, exp_min_val)
+                    self.assertEqual(modules[str(node.target)].max_val, exp_max_val)
+                    inp_counter += 1
+
+                elif node.op == 'call_module' and isinstance(modules[str(node.target)], MinMaxObserver):
+                    # Check min/max values of weight activation layers
+                    assert("weight" in node.name)
+                    exp_min_val, exp_max_val = exp_weight_act_vals[weight_counter]
+                    self.assertEqual(modules[str(node.target)].min_val, exp_min_val)
+                    self.assertEqual(modules[str(node.target)].max_val, exp_max_val)
+                    weight_counter += 1
+
+    def check_orig_and_eq_graphs(self, orig_model, eq_model):
+        """ Given a non-equalized model and an equalized model, check that the
+        graphs are structured in the same way, except the equalized model has
+        additional 'equalization_scale' and 'mul' nodes.
+        """
+        orig_idx = 0
+        orig_nodes = list(orig_model.graph.nodes)
+        orig_modules = dict(orig_model.named_modules(remove_duplicate=False))
+
+        eq_idx = 0
+        eq_nodes = list(eq_model.graph.nodes)
+        eq_modules = dict(eq_model.named_modules(remove_duplicate=False))
+
+        while orig_idx < len(orig_nodes) and eq_idx < len(eq_nodes):
+            if 'equalization_scale' in eq_nodes[eq_idx].name and 'mul' in eq_nodes[eq_idx + 1].name:
+                # Skip the equalization and mul nodes
+                eq_idx += 2
+                continue
+            elif orig_nodes[orig_idx].op != eq_nodes[eq_idx].op:
+                return False
+            elif orig_nodes[orig_idx].op == 'call_module':
+                # Check that the type of call_modules are the same (ex. nn.Linear, MinMaxObserver)
+                orig_node = orig_nodes[orig_idx]
+                eq_node = eq_nodes[eq_idx]
+                if type(orig_modules[orig_node.target]) is not type(eq_modules[eq_node.target]):
+                    return False
+            elif orig_nodes[orig_idx].op == 'call_function':
+                # Check that the call_functions are the same (ex. F.linear)
+                orig_node = orig_nodes[orig_idx]
+                eq_node = eq_nodes[eq_idx]
+                if orig_node.target != eq_node.target:
+                    return False
+
+            eq_idx += 1
+            orig_idx += 1
+
+        return True
+
+    def test_input_weight_equalization_graphs(self):
```

Comment:
Can we use `checkGraphModeFxOp` with an expected node_list and node_occurance? Does testing it this way provide any additional coverage?

Reply:
Yes, I think I could use `checkGraphModeFxOp` with the `expected_node_list` which checks the orders of the occurrences of the nodes - this way we can accurately check the placement of the mul node.

Reply:
I misread the `checkGraphModeFxOp` function and I don't think I'll be able to use it to check the order of occurrences of the node because it runs `prepare_fx` and `convert_fx` without equalization on the model that gets passed in. But, I found that `checkGraphModuleNodes` with an `expected_node_list` also works the same (I have updated the code).